### DISCUSSION TITLE
How to dynamically generate color

### DISCUSSION THREAD

**Discussion by JohannesMisch** [neuefische/questions#495] [Web] [answered]
How to dynamically generate colors for my doughnut chart, using hsl?
The color should be generated for each category that is used inside my devices array.
Thanks in advance

**Comment by doemser** [comment, marked as answer]
I love this function:

```javascript
function getHSLColor(hue) {
  return `hsl(${hue}, 60%, 40%)`
}

```

You could specify the HUE value for each category and then use the function.

Is that what you were asking for?

**Reply by doemser** [reply]
https://codesandbox.io/s/reverent-blackwell-gr8b7i?file=/src/App.js

**Reply by shebtastic** [reply]
I think it's more about mapping enough colors for all entries in the dataset:

```js
//given a dataset with the structure of
const dataset = {
  categories: {
    "Entertainment": 20,
    "Appliance": 7,
  }
}

const rotationStep = 75 //steps in degree

const fillColors = Object.keys(dataset.categories)
  .map((category, index) => `hsl(${0 + index * rotationStep % 360} 100%  50% / 1)`)
const borderColors = Object.keys(dataset.categories)
  .map((category, index) => `hsl(${0 + index * rotationStep % 360} 100%  33% / 1)`)
```

It would of course be nice if the colors would be based off hashes of the strings and still make be part of a useful color palette. But using the positional information is a good start. 😁

**Reply by JohannesMisch** [reply]
Thank you both for your replies. I was looking for @shebtastic answer, but it never hurts to get more useful information. :) I am sure your answer comes in handy in the future @doemser.
